Enable llama2-70b LoRA finetuning

mandy-li opened the pull request:
As title

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/optimum-habana/pr_527). All of your documentation changes will be reflected on that endpoint.

mandy-li commented:
@regisss , this PR is for SynapseAI 1.13 release.

schoi-habana commented on examples/language-modeling/README.md:
```diff
@@ -455,6 +456,65 @@ LOWER_LIST=ops_bf16.txt python3 ../gaudi_spawn.py \
+  --pipelining_fwd_bwd \
+  --throughput_warmup_steps 3 \
+  --lora_rank 4 \
+  --lora_target_modules "q_proj" "v_proj" 'k_proj' 'o_proj' \
```
just a minor thing. q, v are in double quotes and the rest use single quotes

mandy-li replied:
Good point. fixed

regisss commented on examples/language-modeling/README.md:
````diff
@@ -455,6 +456,65 @@ LOWER_LIST=ops_bf16.txt python3 ../gaudi_spawn.py \
+```bash
+PT_HPU_MAX_COMPOUND_OP_SIZE=10 DEEPSPEED_HPU_ZERO3_SYNC_MARK_STEP_REQUIRED=1 \
+python3 ../gaudi_spawn.py --use_deepspeed  --world_size 8  run_lora_clm.py \
+  --model_name_or_path path_to_llama2_70b_ckpt/Llama-2-70b-hf \
````
```suggestion
  --model_name_or_path meta-llama/Llama-2-70b-hf \
```

regisss commented on examples/language-modeling/README.md:
```diff
@@ -455,6 +456,65 @@ LOWER_LIST=ops_bf16.txt python3 ../gaudi_spawn.py \
 
+- Multi-card finetuning of Llama2-70B with DeepSpeed ZeRO-3 optimization and LoRA:
+
+  **Note:** the following command requires Habana DeepSpeed 1.13.0 or later.
```
```suggestion
  > The following command requires Habana DeepSpeed 1.13.0 or later.
```

regisss commented on examples/language-modeling/README.md:
```diff
@@ -321,7 +321,8 @@ python run_clm.py \
 ## PEFT
 
-To run LoRA finetuning and inference. you could use `run_lora_clm.py` as an example. Multi-card examples can be simply adapted to run LoRA finetuning. Here is the CLM example with Llama1-7B and Falcon-40B:
+To run LoRA finetuning, you can use `run_lora_clm.py`. Multi-card examples can be simply adapted to run LoRA finetuning.
+Here are command examples for Llama1-7B, Falcon-40B and Llama2-70B:
```
```suggestion
To run LoRA finetuning, you can use `run_lora_clm.py`.
Here are single-/multi-device command examples for Llama1-7B, Falcon-40B and Llama2-70B:
```
Let's change that too, it sounds a bit redundant.

mandy-li replied:
Agree. addressed

regisss reviewed (approved):
LGTM!

I left a few minor comments. Do we want to merge this before 1.13 is released or right after?

mandy-li commented on examples/language-modeling/README.md:
````diff
@@ -455,6 +456,65 @@ LOWER_LIST=ops_bf16.txt python3 ../gaudi_spawn.py \
+        "contiguous_gradients": false
+    }
+}
+```
````
Agree. Changed

mandy-li commented:
> LGTM!
> 
> I left a few minor comments. Do we want to merge this before 1.13 is released or right after?

@regisss , Thanks for review and valuable comments! We should merge right after 1.13 is released. I addressed your comments in the new commit. Please check again.

regisss commented on examples/language-modeling/README.md:
```diff
@@ -455,6 +456,43 @@ LOWER_LIST=ops_bf16.txt python3 ../gaudi_spawn.py \
 
+- Multi-card finetuning of Llama2-70B with DeepSpeed ZeRO-3 optimization and LoRA:
+
+  The following command requires Habana DeepSpeed 1.13.0 or later.
```
```suggestion
  > The following command requires Habana DeepSpeed 1.13.0 or later.
```
